Orange/widgets/classify/owclassificationtree.py

```diff
@@ -1,24 +1,21 @@
 from Orange.data import Table
 from Orange.classification.tree import TreeLearner, TreeClassifier
-from Orange.widgets import widget, gui
+from Orange.widgets import gui
 from Orange.widgets.settings import Setting
-from Orange.widgets.utils.owlearnerwidget import OWProvidesLearner
-from Orange.widgets.utils.sql import check_sql_input
+from Orange.widgets.utils.owlearnerwidget import OWProvidesLearnerI
 
 
-class OWClassificationTree(OWProvidesLearner, widget.OWWidget):
+class OWClassificationTree(OWProvidesLearnerI):
     name = "Classification Tree"
     icon = "icons/ClassificationTree.svg"
     description = "Classification tree algorithm with forward pruning."
     priority = 30
 
     LEARNER = TreeLearner
+    OUTPUT_MODEL_CLASS = TreeClassifier
```

Comment:
Learners should define `__returns__`, which is a type of the returned model, so e.g. `TreeClassifier == TreeLearner.__returns__`.